### DISCUSSION TITLE
TypeError: MySensor: No constructor defined!

### DISCUSSION THREAD

**Discussion by JiangGuotai** [mitsuba-renderer/mitsuba3#347] [Q&A] [answered]
I wanna define a sensor class using template in python, but when I use the code:
```
class MySensor(mi.ProjectiveCamera):
    def __init__(self, props):
        mi.ProjectiveCamera.__init__(self, props)
```
there will be a error: No constructor defined!

I can only use 
```
class MySensor(mi.Sensor):
    def __init__(self, props):
        mi.Sensor.__init__(self, props)
```
but in C++, ProjectiveCamera is used as a parent class, so... what can i do to use ProjectiveCamera as a parent class in python?

**Comment by njroussel** [comment, marked as answer]
The reason why it only works with `mi.Sensor` is that we only defined Python bindings on the top-level base classes for every plugin. This is done in `src/render/python/sensor_v.cpp`, we define a `pybind11` trampoline ([more information](https://pybind11.readthedocs.io/en/stable/advanced/classes.html#overriding-virtual-functions-in-python)).

If you wanted to define a Python class which derives from `ProjcetiveCamera`. I believe you would need to add a similar trampoline to the existing one for `Sensor` but for `ProjectiveCamera`.
Another approach would be to re-implement the `ProjectiveCamera` in Python, and then have your new `Sensor` inherit from that one instead.

**Comment by JiangGuotai** [comment]
ok, I am trying to re-implement the `ProjectiveCamera`, thank you for the reply.
